GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts

```diff
@@ -120,47 +149,214 @@ namespace gdjs {
+            object.getDrawableY() - cameraTopLeftY,
+          ];
+          if (object instanceof gdjs.SpriteRuntimeObject) {
+            originPoint = object.getPointPosition('origin');
+            originPoint[0] = originPoint[0] - cameraTopLeftX;
+            originPoint[1] = originPoint[1] - cameraTopLeftY;
+          }
+
+          debugDraw.line.color = 0xff0000;
+          debugDraw.fill.color = 0xff0000;
+          debugDraw.drawCircle(originPoint[0], originPoint[1], 3);
+
+          if (showPointsNames) {
+            if (!renderedObjectTextPoints.points['origin']) {
+              renderedObjectTextPoints.points['origin'] = new PIXI.Text(
+                'Origin',
+                {
+                  fill: '#ff0000',
+                  fontSize: 12,
+                }
+              );
+            }
+            renderedObjectTextPoints.points['origin'].position.set(
+              originPoint[0],
+              originPoint[1]
+            );
+
+            // Mark the text points as rendered
+            renderedObjectTextPoints.wasRendered = true;
+            this._pixiContainer.addChild(
+              renderedObjectTextPoints.points['origin']
+            );
+          }
+
+          // Draw custom point
+          if (showCustomPoints && object instanceof gdjs.SpriteRuntimeObject) {
+            if (!object._animationFrame) continue;
+            for (let customPointName in object._animationFrame.points[
+              'items'
+            ]) {
+              let customPoint: [float, float];
+              customPoint = object.getPointPosition(customPointName);
+
+              if (showPointsNames) {
+                if (!renderedObjectTextPoints.points[customPointName]) {
+                  renderedObjectTextPoints.points[
+                    customPointName
+                  ] = new PIXI.Text(customPointName, {
+                    fill: '#0000ff',
+                    fontSize: 12,
+                  });
+                }
+
+                renderedObjectTextPoints.points[customPointName].position.set(
+                  customPoint[0] - cameraTopLeftX,
+                  customPoint[1] - cameraTopLeftY
+                );
+
+                // Mark the text points as rendered
+                renderedObjectTextPoints.points[
+                  customPointName
+                ].wasRendered = true;
+                this._pixiContainer.addChild(
+                  renderedObjectTextPoints.points[customPointName]
+                );
+              }
+
+              debugDraw.line.color = 0x0000ff;
+              debugDraw.fill.color = 0x0000ff;
+              debugDraw.drawCircle(
+                customPoint[0] - cameraTopLeftX,
+                customPoint[1] - cameraTopLeftY,
+                3
+              );
+            }
+          }
+
+          // After we've iterated on all objects, maybe some were destroyed. So clean their text points:
+          for (let objectID in this._debugDrawRenderedObjectsPoints) {
```

Comment:
This loop is done inside the `for (let i = 0; i < instances.length; i++) {`, which means that for every instance you clean the points, which is a performance hit/unecessary complexity.

When a function is becoming too complex, it's easy to make mistakes like this. Question every indentation (do I really need this?) and break a function in smaller functions to help you manager in your head what a function is doing.
I'm fixing this.